### PR TITLE
:bug: fix make build-installer command to remove CRD duplication

### PR DESCRIPTION
**Pull request by mateusoliveira43**
## Why the changes were made

Noticed that when running `make build-installer` in my project, CRDs were being included twice. This PR fixes this.

## How the changes were made

Searched for the commit that introduced logic (https://github.com/kubernetes-sigs/kubebuilder/commit/f39e53e65a6661f1e6e63fe722f8267466c8363d#diff-e5d6d4f9d67174f1f3764674eb786688353c77cef8514e24d1323e55e5a4a7e8) to study it. 

I believe the `if` statement is not necessary, because if you run `kubebuilder init`, you will have no `config/crd` folder; but once you run `kubebuilder create api`, `../crd` is uncommented in `resources` field of `config/default/kustomization.yaml` file:
```diff

 resources:
-#- ../crd
+- ../crd
 - ../rbac
```
Which, will trigger CRDs being included when running `kustomize build config/default`.

After removing `if` block, ran `make generate`. 

## How to test the changes made

Confirm the changes do not break any use case.

### CODE REVIEW

File: testdata/project-v4/Makefile [hunk 1]
```diff
@@ -119,10 +119,6 @@ docker-buildx: ## Build and push docker image for the manager for cross-platform
 build-installer: manifests generate kustomize ## Generate a consolidated YAML with CRDs and deployment.
 	mkdir -p dist
 	echo "---" > dist/install.yaml # Clean previous content
-	@if [ -d "config/crd" ]; then \
-		$(KUSTOMIZE) build config/crd > dist/install.yaml; \
-		echo "---" >> dist/install.yaml; \
-	fi
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
 	$(KUSTOMIZE) build config/default >> dist/install.yaml
```
Comment (lukas016) [inline testdata/project-v4/Makefile:123]:
```suggestion
	$(KUSTOMIZE) build config/default > dist/install.yaml
```

File: testdata/project-v4/Makefile [hunk 2]
```diff
@@ -119,10 +119,6 @@ docker-buildx: ## Build and push docker image for the manager for cross-platform
 build-installer: manifests generate kustomize ## Generate a consolidated YAML with CRDs and deployment.
 	mkdir -p dist
 	echo "---" > dist/install.yaml # Clean previous content
```
Comment (lukas016) [inline testdata/project-v4/Makefile:121]: you don't need this line with suggestion bottom.

Comment (mateusoliveira43) [reply testdata/project-v4/Makefile:121]: agreed, will remove in next commit. Thanks!

**Review by lukas016 — APPROVED** [review]
LGTM

**Review by camilamacedo86 — APPROVED** [review]
@mateusoliveira43 

That is terrific amazing 🥇 

/lgtm
/approved

/ok-to-test

**Comment by camilamacedo86** [comment]
It is passing in the tests the issue here is faced in the legacy check which is removed and no longer exist on master branch so for we do not ask for the author have extra effort I am forcing the merge which seems very safe in this case.

**Pull request merged**